src/main/scala/org/constellation/storage/CheckpointService.scala

```diff
@@ -135,6 +138,62 @@ class CheckpointService[F[_]: Sync: LiftIO](
       (s: String) => ???
     )
 
+  def accept(checkpoint: FinishedCheckpoint)(implicit dao: DAO): F[Unit] = {
+
+    val peers = dao.readyPeers.map { ready =>
+      val filtered = ready.filter(t => checkpoint.facilitators.contains(t._1))
+      (if (filtered.isEmpty) ready else filtered)
+        .map(p => PeerApiClient(p._1, p._2.client))
+        .toList
+    }
+
+    (dao.nodeState, checkpoint.checkpointCacheData.checkpointBlock) match {
+      case (_, None) => Sync[F].raiseError(MissingCheckpointBlockException)
+      case (NodeState.Ready, Some(cb)) =>
+        for {
+          _ <- syncPending(pendingAcceptanceFromOthers, cb)
+          _ <- LiftIO[F].liftIO(resolveMissingParents(cb, peers))
+          _ <- accept(checkpoint.checkpointCacheData)
+        } yield ()
+      case (NodeState.DownloadCompleteAwaitingFinalSync, Some(_)) =>
+        LiftIO[F].liftIO(dao.snapshotService.syncBufferAccept(checkpoint.checkpointCacheData))
+      case (_, Some(_)) => Sync[F].raiseError(PendingDownloadException(dao.id))
+    }
+  }
+
+  def resolveMissingParents(cb: CheckpointBlock, peers: IO[List[PeerApiClient]], depth: Int = 1)(
+    implicit dao: DAO
+  ): IO[List[CheckpointCache]] = {
```

Comment:
`F[List[CheckpointCache]]`. please don't mix `IO` with `F` in one class

Reply:
It was temporary to not rewrite everything on early stage.